### PR TITLE
Mjpeg dec

### PR DESCRIPTION
**Pull request by kalistratovag**

### CODE REVIEW

File: modules/videoio/src/cap_mjpeg_decoder.cpp [hunk 1]
```diff
@@ -40,13 +40,733 @@
...
+double MotionJpegCapture::getProperty(int property) const
+{
+    switch(property)
+    {
+        case CAP_PROP_POS_FRAMES:
+            return (double)getFramePos();
+        case CAP_PROP_POS_AVI_RATIO:
+            return double(getFramePos())/m_mjpeg_frames.size();
+        case CAP_PROP_FRAME_WIDTH:
+            return 0;
```
Comment (vpisarev) [inline modules/videoio/src/cap_mjpeg_decoder.cpp]: can you please implement FRAME_WIDTH, FRAME_HEIGHT and FPS properties? They are used quite often by users?

File: modules/videoio/src/cap_mjpeg_decoder.cpp [hunk 2]
```diff
@@ -40,13 +40,733 @@
...
+    if(m_frame_iterator != m_mjpeg_frames.end())
+    {
+        std::vector<char> data = readFrame(m_frame_iterator);
+
+        if(data.size())
+        {
+            m_current_frame = imdecode(data, CV_LOAD_IMAGE_ANYDEPTH | CV_LOAD_IMAGE_COLOR);
+        }
+
+        output_frame.getMatRef() = m_current_frame;
```
Comment (vpisarev) [inline modules/videoio/src/cap_mjpeg_decoder.cpp]: it may be not a good solution. Please, use m_current_frame.copyTo(output_frame) instead.

**Comment by vpisarev** [comment]
@kalistratovag, thanks! overall, it looks very good! please, fix the 2 issues mentioned above. Also, can you please "squash" (combine) all the commits into one? Ask Maksim on how to do it.

**Comment by vpisarev** [comment]
thank you! :+1:

**Comment by vpisarev** [comment]
@mshabunin, looks like the recent fix in OpenCVConfig broke the Linux build

**Comment by mshabunin** [comment]
@vpisarev, it is OK now. Previous build used commit 369a3d1 for merge, one before the fix.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**